core/src/main/scala/org/apache/spark/deploy/history/EventLogFileCompactor.scala

```diff
@@ -221,5 +222,15 @@ private class CompactedEventLogFileWriter(
     hadoopConf: Configuration)
   extends SingleEventLogFileWriter(appId, appAttemptId, logBaseDir, sparkConf, hadoopConf) {
 
-  override val logPath: String = originalFilePath.toUri.toString + EventLogFileWriter.COMPACTED
+  override val shouldCompress = EventLogFileWriter.codecName(originalFilePath).isDefined
+
+  override val compressionCodec =
+    if (shouldCompress) {
+      val originalCodecName = EventLogFileWriter.codecName(originalFilePath).getOrElse("")
```

Comment:
When do we hit `.getOrElse("")`? `.getOrElse("")` looks like no-op to me because we have a guard condition, `shouldCompress`.